modules/account-lib/src/coin/dot/utils.ts

```diff
@@ -109,8 +109,7 @@ export class Utils implements BaseUtils {
     tx: string | UnsignedTransaction,
     options: { metadataRpc: string; registry: TypeRegistry },
   ): TransferArgs {
-    const { metadataRpc, registry } = options;
-    registry.setMetadata(createMetadata(registry, metadataRpc));
```

Comment:
Why have we changed the setMetadata here? Based on the comments, it seems crucial that we set metadata before signing

Reply:
Metadata is set when the singleton is created as in:  
metadataRpc: material.metadata as `0x${string}`,
That is why all tests are passing even if we are not setting metadata explicitly, we need to set metadata  again whenever we make a change in the material.

Reply:
ah ok, good to know!